File: java/src/ch4/lc406_minimum_size_subarray_sum/MinimumSizeBF.java
package ch4.lc406_minimum_size_subarray_sum;

public class MinimumSizeBF {

    public static void main(String[] args) {
        System.out.println(new MinimumSizeBF().minimumSize(new int[]{2, 3, 1, 2, 4, 3}, 7));
    }

    public int minimumSize(int[] nums, int s) {
        int minLength = Integer.MAX_VALUE;
        for (int start = 0; start < nums.length; start++) {
            for (int end = start; end < nums.length; end++) {
                int sumOfSubarray = 0;
                for (int i = start; i <= end; i++) {
                    sumOfSubarray += nums[i];
                }
                if (sumOfSubarray >= s) {
                    minLength = Math.min(minLength, end - start + 1);
                }
            }
        }
        if (minLength == Integer.MAX_VALUE) {
            return -1;
        }
        return minLength;
    }

}

File: python/ch4-2pointer-1dir/lc406-minimum-size-subarray-sum/minimum_size_bf.py
class Solution:

    def minimum_size(self, nums, s):
        min_length = float('inf')
        for start in range(len(nums)):
            for end in range(start, len(nums)):
                sum_of_subarray = 0
                for i in range(start, end + 1):
                    sum_of_subarray += nums[i]
                if sum_of_subarray >= s:
                    min_length = min(min_length, end - start + 1)
        if min_length == float('inf'):
            return -1
        return min_length


solution = Solution()
print(solution.minimum_size([2,3,1,2,4,3], 7))
